packages/amplify-cli-logger/src/AmplifyLogger.ts

```diff
@@ -0,0 +1,87 @@
+import winston, { Logger, format } from 'winston';
+import winstonDailyRotateFile from 'winston-daily-rotate-file';
+import { constants } from './constants';
+import { IAmplifyLogger } from './IAmplifyLogger';
+import { JSONUtilities } from 'amplify-cli-core';
+import { getLogFilePath, getLocalLogFilePath, getLogAuditFilePath, getLocalAuditLogFile } from './getLogFilePath';
+import { LocalProjectData, LogPayload, LogErrorPayload } from './Types';
+import { Redactor } from './Redactor';
+
+export class AmplifyLogger implements IAmplifyLogger {
+  logger: Logger;
+  format: winston.Logform.Format;
+  localProjectData!: LocalProjectData;
+  disabledAmplifyLogging: boolean = !!process.env.AMPLIFY_CLI_DISABLE_LOGGING;
+  options: any = {
+    mode: 0o640,
```

Comment:
To help with future maintenance, when adding config in code that is opaque in nature (such as a hex value) is to add a comment on what this represents and *why* we are choosing this value. Can you add a note here?